simulator/mips/mips_instr.h

```diff
@@ -305,7 +309,29 @@ class FuncInstr
             if ( _is_jump_taken)
                 new_PC += sign_extend( v_imm) << 2;
         }
-
+        
+        void execute_clz()   
+        {
+            v_dst = 0;
+            for ( uint32_t i = 0x80000000; i > 0; i >>= 1)
+            {
+		        if ( v_src1 & i)
+		            break;  
+	            v_dst++;  
+	        }
+	    }
+        
+        void execute_clo()   
```

Comment:
I would avoid code duplication by creating some common function that counts bits. Than, code would be simpler:
```
uint32 count_ones(uint32 value) const { /* ... */ }
// ...
void execute_clo() { v_dst = count_ones( v_src1); }
void execute_clz() { v_dst = sizeof(uint32) * 8 - count_ones( v_src1); }
```